src/main/java/com/mixpanel/android/mpmetrics/SystemInformation.java

```diff
@@ -122,14 +122,13 @@ public String getCurrentNetworkOperator() {
         return ret;
     }
 
-
     public Boolean isWifiConnected() {
         Boolean ret = null;
 
         if (PackageManager.PERMISSION_GRANTED == mContext.checkCallingOrSelfPermission(Manifest.permission.ACCESS_NETWORK_STATE)) {
             ConnectivityManager connManager = (ConnectivityManager) this.mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
-            NetworkInfo wifiInfo = connManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
-            ret = wifiInfo.isConnected();
+            NetworkInfo networkInfo = connManager.getActiveNetworkInfo();
```

Comment:
`getNetworkInfo` is deprecated in API 23. `getActiveNetworkInfo` returns `Returns details about the currently active default data network`. If a user is both on 4G and WiFi, the default one should be Wifi.